File: Object Oriented Programming/learning/classes_and_objects/exercises/py/ex-3.py
"""
Write a program to print the area and perimeter of a triangle having sides 
of 3, 4 and 5 units by creating a class named 'Triangle' with a function to 
print the area and perimeter.
"""

# define class
class Triangle:
    # initialize side variables
    side1, side2, side3 = 0.0, 0.0, 0.0

    @staticmethod
    def printArea(side1, side2, side3):
        area = (side1+side2+side3)/2.0
        print(f"Area: {area}\n"
              f"Perimeter: {(side1+side2+side3)}")
    
# create an object of the class
Triangle.printArea(3, 4, 5)


"""
The printArea function is defined as a static method using the @staticmethod decorator. 
This allows you to call the method without creating an instance of the class.
"""

File: Object Oriented Programming/learning/classes_and_objects/exercises/cpp/ex-3.cpp
/*
Write a program to print the area and perimeter of a triangle having sides 
of 3, 4 and 5 units by creating a class named 'Triangle' with a function to 
print the area and perimeter.
*/

#include <iostream>
using namespace std;

// define class
class Triangle {
    private:
        double side1, side2, side3;
    
    public:
        void print_area(
            double sid1, double sid2, double sid3
        ) {
            double area = (sid1+sid2+sid3)/2.0;
            cout<<"Area: "<<area<<endl;
            cout<<"Perimeter: "<<(sid1+sid2+sid3)<<endl;
        }
};

int main() {
    // create an object and assign values
    Triangle tri;
    tri.print_area(3, 4, 5);

    return 0;
}


